database.go

```
@@ -1,37 +1,81 @@
 package main
 
 import (
-	"database/sql"
+	"errors"
+	"fmt"
+	"strings"
 )
 
-var db *sql.DB
+var (
+	ErrDBConnStrParseNoSep = errors.New("Missing ':' in connection string")
+	ErrDBConnStrParseNotImplemented = func(dbName string) error {
+		return errors.New(fmt.Sprintf("The database '%s' is not implemented", dbName))
+	}
+)
 
-func LoadDatabase(dbFile string) error {
-	var err error
-	db, err = sql.Open("sqlite3", dbFile)
+type Database interface {
+	CommentService
+}
+
+func parseConnectionStr(connectionStr string) (dbName string, dbParams map[string]interface{}, err error) {
+
+	// Derive the database name dbName (sqlite, postgres, mongo etc.)
+	dbNameSep := strings.Index(connectionStr, ":")
+	if dbNameSep == -1 {
+		err = ErrDBConnStrParseNoSep
+		return
+	}
+
+	dbName = strings.TrimSpace(connectionStr[:dbNameSep])
+
+	// Get the parameters to initialize the database
+	dbParams = make(map[string]interface{})
+	for _, param := range strings.Split(connectionStr[dbNameSep+1:], ";") {
+		paramSep := strings.Index(param, "=")
+		if paramSep != -1 {
+			key := strings.TrimSpace(param[:paramSep])
+			if len(key) > 0 {
```

This is probably a misconfiguration. We'll want to return an error.